File: mc/src/mcfinc.c
/* $Id: mcfinc.c,v 1.2 1995/02/24 10:22:52 axel Rel $
 */

#include <math.h>
#include <errno.h>
#include <arx_def.h>

double cterm (double interest, double futurevalue, double presentvalue)
{
return log (futurevalue/presentvalue) / log (1+interest);
}

double ddb (double cost, double salvage, double life, double period)
{
int	i, n = (int) (period+.5);
double	d, val = cost;

if (n < 1 || n > (int)(life+.5))
	{
	errno = ERANGE;
	return HUGE_VAL;
	}
for (i = 1; i <= n; i++)
	{
	d	= (val*2)/life;
	val	-= d;
	}
val += d;
return val > salvage ? min (d, val-salvage) : 0.;
}

double fv (double annuity, double interestrate, double period)
{
return annuity * (pow(interestrate+1.,period) - 1.) / interestrate;
}

double pmt (double principal, double interest, double period)
{
return principal * (interest / (1. - pow(interest+1.,-period)));
}

double pv (double annuity, double interestrate, double period)
{
return annuity * (1.-pow(1.+interestrate,-period)) / interestrate;
}

double rate (double futurevalue, double presentvalue, double n)
{
return pow (futurevalue / presentvalue, 1./n) - 1.;
}

double sln (double cost, double salvage, double life)
{
return (cost-salvage) / life;
}

double syd (double cost, double salvage, double life, double period)
{
return (cost-salvage) * (life-period+1.) / (life * (life+1.)/2.);
}

double term (double payment, double interest, double futurevalue)
{
return log (1.+(futurevalue*interest/payment)) / log (1.+interest);
}

File: mc/man/mcfinc.r
.Id $Id: mcfinc.r,v 1.3 1997/06/26 18:46:28 axel Exp $
.SH "Financial functions"
.IP CTERM(INTerest;FutureValue;PresentValue) 3
The number of compounding periods required for an
investment of PV (present value) to grow to FV (future
value), if the investment is earning INT interest per
compounding period.
.IP DDB(Cost;Salvage;Life;Period) 3
Accelerated depreciation value for an asset, given the
initial cost, the life expectancy, the end value and the
depreciation period.  The depreciation is calculated
using the double declining balance method.
.IP "FV(Annuity;interest rate;period)" 3
Returns the future value corresponding to  an  annuity
occurring over the specified period.  Reverse of the
above.
.br
ex. FV(700;0.0925;30) = 99979.77
.IP PMT(Principal;interest;period) 3
Returns the amount of an annuity equal to the
principal amortized over the period.
.br
ex. PMT(75000;0.0925;30) = 7462.61
.IP "PV(Annuity;interest rate;period)" 3
Returns the present value corresponding  to  an  annuity
occurring over the specified period.  Reverse of the
above.
.br
ex. PV(700;0.0925;30) = 7035.08
.IP RATE(FutureValue;PresentValue;N) 3
Interest rate required for an investment of PVAL to be
worth FVAL, in N compounding periods.
.IP SLN(Cost;Salvage;Life) 3
Straight line depreciation of an asset over ONE period
of LIFE.
.IP SYD(Cost;Salvage;Life;Period) 3
Depreciation of an asset using the Sum-of-Years
depreciation method.
.IP TERM(Payment;Interest;FutureValue)
Number of periods required to accumulate an investment
of FVAL making regular payments of PMT and accruing
interest at the rate of INT per period.
